src/docs/guides/23-how-to-use-custom-pricing/index.md

```diff
@@ -0,0 +1,793 @@
+`CheckoutPage` has to be modified a bit. Add the following lines after
+the current imports at the top of the `CheckoutPage.js` file:
+
+```js
+import { nightsBetween, daysBetween } from '../../util/dates';
+import { types as sdkTypes } from '../../util/sdkLoader';
+
+const { Money } = sdkTypes;
+```
+
+Then add the following method to the `CheckoutPageComponent` class:
+
+```js
+/**
+ * Constructs a request params object that can be used when creating bookings
+ * using custom pricing.
+ * @param {} params An object that contains bookingStart, bookingEnd and listing
+ * @return a params object for custom pricing bookings
+ */
+customPricingParams(params) {
+  const { bookingStart, bookingEnd, listing, ...rest } = params;
+  const { amount, currency } = listing.attributes.price;
+
+  const unitType = config.bookingUnitType;
+  const isNightly = unitType === LINE_ITEM_NIGHT;
+
+  const quantity = isNightly
+    ? nightsBetween(bookingStart, bookingEnd)
+    : daysBetween(bookingStart, bookingEnd);
+
+  const bookingLineItem = {
+    code: unitType,
+    unitPrice: new Money(amount, currency),
+    quantity,
+  };
+
+  const lineItems = [bookingLineItem];
+
+  return {
+    listingId: listing.id,
+    bookingStart,
+    bookingEnd,
+    lineItems,
+    ...rest,
+  };
+}
+```
+
+When it's in place the method can be used to resolve new kind of request
+params.
+
+In the `loadInitialData` method:
+
+```js
+fetchSpeculatedTransaction(
+  this.customPricingParams({
+    listing,
+    bookingStart: bookingStartForAPI,
+    bookingEnd: bookingEndForAPI,
+  })
+);
+```
+
+And in `handleSubmit`:
+
+```js
+const requestParams = this.customPricingParams({
+  listing: this.state.pageData.listing,
+  cardToken,
+  bookingStart: speculatedTransaction.booking.attributes.start,
+  bookingEnd: speculatedTransaction.booking.attributes.end,
+});
+```
+
+## 4. Add cleaning fee to the listing page UI
+
+Next step is to modify the listing page UI so that a user can add the
+cleaning fee to the order. This guide expects that the cleaning fee
+price is stored in listing public data in an object with two keys:
+`amount` and `currency`. The `amount` attribute holds the price in sub
+units whereas `currency` holds the currency code. Like this for a
+cleaning fee of 20€:
+
+```js
+publicData: {
+  cleaningFee: { amount: 2000, currency: 'EUR' }
+}
+```
+
+Begin with making a few changes to the `util/types.js` and `en.json`
+files (or some other transaltion file in case English is not the
+language of your marketplace). In `util/types.js` add a new line item
+code and add it to the array of line items:
+
+```js
+export const LINE_ITEM_CLEANING_FEE = 'line-item/cleaning-fee';
+
+const LINE_ITEMS = [
+  LINE_ITEM_NIGHT,
+  LINE_ITEM_DAY,
+  LINE_ITEM_UNITS,
+  LINE_ITEM_CUSTOMER_COMMISSION,
+  LINE_ITEM_PROVIDER_COMMISSION,
+  LINE_ITEM_CLEANING_FEE,
+];
+```
+
+To the `en.json` translation file, add the following new translation
```

Comment:
This feels a bit out of context. 
1) You could maybe add notes next to correct component examples: 
```markdown
> Note: remember to add the translation key to `en.json` file: 
> `"BookingDatesForm.cleaningFee": "Add cleaning",`
```

2. or you could elaborate more here. Something like 'We are going to show "Add cleaning" checkbox on BookingDatesForm and "Cleaning fee" line-item in BookingBreakdown component.'